mmcv/runner/ipu/hooks.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from mmcv.utils import TORCH_VERSION, digit_version
+from mmcv.runner.hooks import HOOKS, OptimizerHook, LrUpdaterHook
+
+
+def wrap_lr_update_hook(lr_hook_class):
+    """A wrapper function to wrap any subclass of LrUpdaterHook
+    IPU needs extra operations to upload optimizer setting,
+    this wrapper will override function(_set_lr) of a subclass of
+    LrUpdaterHook.
+    """
+    assert issubclass(lr_hook_class, LrUpdaterHook)
+
+    class ipu_lr_hook_class(lr_hook_class):
+        def _set_lr(self, runner, *args, **kwargs):
+            result = super()._set_lr(runner, *args, **kwargs)
+            assert result is None  # _set_lr should return nothing
```

Comment:
Why do we need to assert that the result is None? Should we directly call `super()._set_lr(runner, *args, **kwargs)` and return nothing?